benchmarks/prune_test.go

```diff
@@ -0,0 +1,83 @@
+package benchmarks
+
+import (
+	"fmt"
+	"os"
+	"runtime"
+	"testing"
+
+	db "github.com/tendermint/tm-cmn/db"
+)
+
+type pruningstrat struct {
+	keepEvery, keepRecent int64
+}
+
+// To test effect of pruning strategy, we must measure time to execute many blocks
+// Execute 30000 blocks with the given IAVL tree's pruning strategy
+func runBlockChain(b *testing.B, prefix string, keepEvery int64, keepRecent int64, keyLen, dataLen int) {
+	// prepare a dir for the db and cleanup afterwards
+	dirName := fmt.Sprintf("./%s-db", prefix)
+	defer func() {
+		err := os.RemoveAll(dirName)
+		if err != nil {
+			b.Errorf("%+v\n", err)
+		}
+	}()
+
+	runtime.GC()
+
+	// always initialize tree with goleveldb as snapshotDB and memDB as recentDB
+	snapDB := db.NewDB("test", "goleveldb", dirName)
+	defer snapDB.Close()
+
+	// var mem runtime.MemStats
```

Comment:
leftover comments

Reply:
currently in process of getting this to work correctly. will remove once i've fixed this